src/debug/HyprCtl.cpp

```diff
@@ -949,11 +950,29 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
     const std::string GPUINFO = execAndGet("lspci -vnn | grep VGA");
 #endif
     result += "GPU information: \n" + GPUINFO;
-    if (GPUINFO.contains("NVIDIA") && std::filesystem::exists("/proc/driver/nvidia/version"))
-        result += execAndGet("cat /proc/driver/nvidia/version | grep NVRM");
+    if (GPUINFO.contains("NVIDIA") && std::filesystem::exists("/proc/driver/nvidia/version")) {
+        std::ifstream file("/proc/driver/nvidia/version");
+        std::string   line;
+        if (file.is_open()) {
+            while (std::getline(file, line)) {
+                if (line.find("NVRM") != std::string::npos) {
+                    result += line;
+                    result += "\n";
+                }
+            }
+            file.close();
+        } else
+            result += "error";
+    }
     result += "\n\n";
 
-    result += "os-release: " + execAndGet("cat /etc/os-release") + "\n\n";
+    std::ifstream file("/etc/os-release");
```

Comment:
this file is not closed properly (will linger until the end of the func), and it should be local probably.

```cpp
if (std::ifstream file("/etc/os-release"); file.is_open()) {
   // ...
}
```

Reply:
Done